graph/articulationpoints.go

```diff
@@ -0,0 +1,62 @@
+// Algorithm to identify articulation points in a connected graph
+// You can read more about articulation points here: https://en.wikipedia.org/wiki/Biconnected_component
+// and the implementation and explanation here: https://cptalks.quora.com/Cut-Vertex-Articulation-point
+
+package graph
+
+import "github.com/TheAlgorithms/Go/math/min"
+
+var (
+	time               int    // time variable to keep track of the time of discovery_time of a vertex
+	is_ap              []bool // boolean array to mark articulation points
+	visited            []bool // boolean array to mark visited vertices
+	child_cnt          []int  // array to store the number of children of a vertex
+	discovery_time     []int  // array to store the time of discovery of a vertex
+	earliest_discovery []int  // array to store the earliest discovered vertex reachable from a vertex
+)
```

Comment:
Avoid global variables. Do you really have to use global variables??

Reply:
@tjgurwara99  So, I tried to do this using a nested function. Basically I wanted to make the helper function nested inside the main function, however because the helper function needs to be recursive I couldn't maneuver around this. The other option is to pass all these global variables as function parameters but that lead to a lot of function parameters. The workaround for this was making a struct and passing the struct in functions. Should I go ahead with the latter approach? Or is there a more subtle way to get this done?

~ Aniruddha

Reply:
I don't see a problem with having a few extra arguments to the function. The struct already exists as well, you can embed and create your own struct but I think that is not ideal for a general purpose graph. So I recommend going with the argument approach but if I remember correctly, only your helper function would have to have all of these parameters and not your main function so, I don't see a problem with the argument approach as the helper function is not an exported function.

Reply:
@tjgurwara99 I've made the changes, have a look and let me know.

~ Aniruddha

Reply:
Hey! Sorry, been a bit busy at work. Will check it tonight and let you know. From a quick glance it seems that the implementation is good. The only issue I have is with your style of commenting - in Go there is an auto documentation friendly commenting style which is used. It seems you are not aware of it. I would suggest adding proper Go styled comments. [Read this for more information](https://go.dev/blog/godoc)